lib/mongo/server/connection_pool/generation_manager.rb

```diff
@@ -30,33 +30,32 @@ def initialize(server:)
 
         attr_reader :server
 
-        def generation(service_id: nil)
-          if service_id
-            unless server.load_balancer?
-              raise ArgumentError, "Generation scoping to services is only available in load-balanced mode, but the server at #{server.address} is not a load balancer"
-            end
-          else
-            if server.load_balancer?
+        def generation(connection_global_id: nil)
+          key = if server.load_balancer?
+            if connection_global_id.nil?
               raise ArgumentError, "The server at #{server.address} is a load balancer and therefore does not have a single global generation"
             end
+            connection_global_id
```

Comment:
If I am reading this correctly, generation must still be based on the service id: https://github.com/mongodb/specifications/blob/master/source/load-balancers/load-balancers.rst#initial-handshake-errors

Reply:
Wow, good catch! Thank you, I missed this moment